units/src/fee.rs

```diff
@@ -160,17 +161,15 @@ impl FeeRate {
 
 /// Computes the ceiling so that the fee computation is conservative.
 impl ops::Mul<FeeRate> for Weight {
-    type Output = Amount;
+    type Output = NumOpResult<Amount>;
```

Comment:
Note to self: this change makes the docs in `result` stale. After #4179 goes in include a change that updates the docs since `NumOpResult` is now used in ops impls that return an `Amount` not just in ops _on_ amounts.

Reply:
Since I had to rebase to fix the review suggestions I just grabbed the new `result` rustdocs from https://github.com/rust-bitcoin/rust-bitcoin/pull/4179#issuecomment-2695745410 and rolled it into the `fee` change in 89751fc12